ddtrace/propagation/http.py

```diff
@@ -885,12 +887,57 @@ def _inject(span_context, headers):
         return headers
 
 
+class _BaggageHeader:
+    @staticmethod
+    def _encode_key(key):
+        safe_characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ" "abcdefghijklmnopqrstuvwxyz" "0123456789" "!#$%&'*+-.^_`|~"
+        encoded = urllib.parse.quote(key, safe=safe_characters)
+        return encoded
+
+    def _encode_value(value):
+        safe_characters = (
+            "ABCDEFGHIJKLMNOPQRSTUVWXYZ" "abcdefghijklmnopqrstuvwxyz" "0123456789" "!#$%&'()*+-./:<>?@[]^_`{|}~"
+        )
+        encoded = urllib.parse.quote(value, safe=safe_characters)
+        return encoded
+
+    @staticmethod
+    def _inject(span_context, headers):
+        baggage_items = span_context._baggage.items()
+        if not baggage_items:
+            return
+
+        header_value = ",".join(
+            f"{_BaggageHeader._encode_key(str(key).strip())}={_BaggageHeader._encode_value(str(value).strip())}"
+            for key, value in baggage_items
+        )
+
+        headers["baggage"] = header_value
+
+    @staticmethod
+    def _extract(headers):
+        header_value = headers.get("baggage")
+        if not header_value:
+            return None
+
+        baggage = {}
+        baggages = header_value.split(",")
+        for key_value in baggages:
+            key, value = key_value.split("=", 1)
```

Comment:
What happens if key_value contains multiple `=` values? Is this behavior called out in the RFC?

Reply:
Keys cannot contain "=" so if there is an "=" it would be encoded.

Reply:
In the case that there is "=" in value (which is allowed), I think the `split` function should split on the first "="

Reply:
Yup, when extracting headers, `key1=foo=bar` should be interpreted as:
```
key:   "key1"
value: "foo=bar"
```